pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -218,33 +221,56 @@ func (m FlavorAssignmentMode) String() string {
 }
 
 type FlavorAssignment struct {
-	Name   kueue.ResourceFlavorReference
-	Mode   FlavorAssignmentMode
-	borrow int64
+	Name      kueue.ResourceFlavorReference
+	Mode      FlavorAssignmentMode
+	FlavorIdx int
+	borrow    int64
+}
+
+func lastAssignmentOutdated(wl *workload.Info, cq *cache.ClusterQueue) bool {
+	return cq.AllocatableResourceGeneration > wl.LastAssignment.ClusterQueueGeneration ||
+		(cq.Cohort != nil && cq.Cohort.AllocatableResourceGeneration > wl.LastAssignment.CohortGeneration)
 }
 
 // AssignFlavors assigns flavors for each of the resources requested in each pod set.
 // The result for each pod set is accompanied with reasons why the flavor can't
 // be assigned immediately. Each assigned flavor is accompanied with a
 // FlavorAssignmentMode.
 func AssignFlavors(log logr.Logger, wl *workload.Info, resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, cq *cache.ClusterQueue, counts []int32) Assignment {
+	if wl.LastAssignment != nil && lastAssignmentOutdated(wl, cq) {
+		wl.LastAssignment = nil
+	}
+
```

Comment:
We will record the last flavor index in `wl.LastAssignment` and start from there next time. When `cq.AllocatebleResourceGeneration` is outdated, we set `wl.LastAssignment = nil` so that we will start from the first flavor.

Reply:
That makes sense. Thank you for clarifying :)